glslc/test/environment.py

```diff
@@ -62,11 +63,16 @@ def write(self, parent):
 
 class File:
     """Specifies a file by name and content."""
-    def __init__(self, name, content):
+    def __init__(self, name, content, readable=True):
         self.name = name
         self.content = content
+        self.readable = readable
 
     def write(self, directory):
         """Writes content to directory/name."""
-        with open(os.path.join(directory, self.name), 'w') as f:
+        full_path = os.path.join(directory, self.name)
+        with open(full_path, 'w') as f:
             f.write(self.content)
+        if not self.readable:
+          access_bits = stat.S_IMODE(os.stat(full_path).st_mode)
+          os.chmod(full_path, access_bits & ~stat.S_IREAD)
```

Comment:
How portable is this?  [Man page](https://docs.python.org/2/library/stat.html) says that this is "Unix V7 synonym for S_IRUSR", whatever that means.